sql/core/v1.2.1/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilters.scala

```diff
@@ -63,55 +64,28 @@ private[sql] object OrcFilters extends OrcFiltersBase {
    */
   def createFilter(schema: StructType, filters: Seq[Filter]): Option[SearchArgument] = {
     val dataTypeMap = schema.map(f => f.name -> f.dataType).toMap
+    val orcFilterConverter = new OrcFilterConverter(dataTypeMap)
     for {
       // Combines all convertible filters using `And` to produce a single conjunction
       conjunction <- buildTree(convertibleFilters(schema, dataTypeMap, filters))
```

Comment:
why not call `buildTree(filters.flatMap(orcFilterConverter.trimUnconvertibleFilters))` here?

Reply:
Changed. However, we still need to keep the `convertibleFilters` method because `OrcScanBuilder` uses it now.

The right thing to do is probably to just make `OrcFilters` a class with `schema` and `filters` passed in as constructor parameters. However, this is an interface change that would require me to change about ~30 call sites to do, so I don't want to add it to this PR. If you think it's a good idea, I'd be happy to open a new PR with that change.